content/source/docs/plugin/framework/validation.html.md

```diff
@@ -0,0 +1,261 @@
+
+All validators will always be run, regardless of whether previous validators returned an error or not.
+
+-> The framework will implement or reference common use case attribute validations, such as string length, in the future.
+
+### Creating Attribute Validators
+
+To create an attribute validator, the [`tfsdk.AttributeValidator` interface](https://pkg.go.dev/github.com/hashicorp/terraform-plugin-framework/tfsdk#AttributeValidator) must be satisfied. For example:
+
+```go
+type stringLengthBetweenValidator struct {
+    Max int
+    Min int
+}
+
+// Description returns a plain text description of the validator's behavior, suitable for a practitioner to understand its impact.
+func (v stringLengthBetweenValidator) Description(ctx context.Context) string {
+    return fmt.Sprintf("string length must be between %d and %d", v.Min, v.Max)
+}
+
+// MarkdownDescription returns a markdown formatted description of the validator's behavior, suitable for a practitioner to understand its impact.
+func (v stringLengthBetweenValidator) MarkdownDescription(ctx context.Context) string {
+    return fmt.Sprintf("string length must be between `%d` and `%d`", v.Min, v.Max)
+}
+
+// Validate runs the main validation logic of the validator, reading configuration data out of `req` and updating `resp` with diagnostics.
+func (v stringLengthBetweenValidator) Validate(ctx context.Context, req tfsdk.ValidateAttributeRequest, resp *tfsdk.ValidateAttributeResponse) {
+    // types.String must be the attr.Value produced by the attr.Type in the schema for this attribute
+    // for generic validators, use
+    // https://pkg.go.dev/github.com/hashicorp/terraform-plugin-framework/tfsdk#ConvertValue
+    // to convert into a known type.
+    var str types.String
+    diags := tfsdk.ValueAs(ctx, req.AttributeConfig, &str)
+    resp.Diagnostics.Append(diags...)
+    if diags.HasError() {
+        return
+    }
+
+    if str.Unknown || str.Null {
+        return
+    }
+
+    strLen := len(str)
+
+    if strLen < v.Min || strLen > v.Max {
+        resp.Diagnostics.AddAttributeError(
+            req.AttributePath,
+            "Invalid String Length",
+            fmt.Sprintf("String length must be between %d and %d, got: %d.", v.Min, v.Max, strLen),
+        )
+
+        return
+    }
+}
+```
+
+Optionally and depending on the complexity, it may be desirable to also create a helper function to instantiate the validator. For example:
+
+```go
+func stringLengthBetween(minLength int, maxLength int) stringLengthBetweenValidator {
+    return stringLengthBetweenValidator{
+        Max: maxLength,
+        Min: minLength,
+    }
+}
+```
+
+## Type Validation
+
+Providers that contain common attribute values with consistent validation rules may wish to create a custom type to simplify schemas. When validation is implemented on a type, it is not necessary to declare the same validation on the attribute, although additional validations can be supplied in that manner. For example:
```

Comment:
```suggestion
You may want to create a custom type to simplify schemas if your provider that contains common attribute values with consistent validation rules. When you implement validation on a type, you do not need to declare the same validation on the attribute, but you can supply additional validations in that manner. For example:
```

Reply:
Just removing an extra word from the suggestion 😄 

```suggestion
You may want to create a custom type to simplify schemas if your provider contains common attribute values with consistent validation rules. When you implement validation on a type, you do not need to declare the same validation on the attribute, but you can supply additional validations in that manner. For example:
```